text/0510-dynamodb-global-table.md

```diff
@@ -0,0 +1,1179 @@
+    },
+    {
+      region: 'us-east-2',
+    },
+  ],
+});
+```
+
+__NOTE:__
+
+* We automatically deploy a replica to the stack region. User only needs to specify additional replication regions.
+* User can add as many replicas when creating the table. But, after that you can only add/remove a single replica
+in a stack update.
+
+#### Global Secondary Indexes
+
+You can add global secondary indexes(GSIs) to your global table. These will be the same for each replica of the
+table.
+
+```typescript
+new GlobalTable(tableStack, 'GlobalTable', {
+  tableName: 'FooTable',
+  partitionKey: {
+    name: 'FooHashKey',
+    type: AttributeType.STRING,
+  },
+  sortKey: {
+    name: 'FooRangeKey',
+    type: AttributeType.STRING,
+  },
+  globalSecondaryIndex: [{
+    indexName: 'UniqueGsiName',
+    partitionKey: {
+      name: 'FooRangeKey',
+      type: AttributeType.STRING,
+    },
+  }],
+  replicas: [{
+    region: 'us-east-1',
+  }],
+});
+```
+
+You can add global secondary index with `addGlobalSecondaryIndex` method.
+
+```typescript
+globalTable.addGlobalSecondaryIndex({
+  indexName: 'UniqueGsiName',
+  partitionKey: { name: 'FooRangeKey', type: AttributeType.STRING },
+});
+```
+
+You can also allocate capacities for your GSIs:
+
+* You only need to provide write and read capacity for GSIs where you want it to be different than capacity specified
+for the global table. If not specified, it uses the same value as that of the global table.
+
+  ```typescript
+  new GlobalTable(tableStack, 'GlobalTable', {
+    tableName: 'FooTable',
+    partitionKey: {
+      name: 'FooHashKey',
+      type: AttributeType.STRING,
+    },
+    sortKey: {
+      name: 'FooRangeKey',
+      type: AttributeType.STRING,
+    },
+    billingMode: BillingMode.PROVISIONED,
+    write: Capacity.autoscaled({ max: 70 }),
+    read: Capacity.fixed(40),
+    globalSecondaryIndex: [{
+      indexName: 'UniqueGsiName',
+      partitionKey: {
+        name: 'FooRangeKey',
+        type: AttributeType.STRING,
+      },
+      write: Capacity.autoscaled({ max: 90 }),
+      read: Capacity.autoscaled({ max: 60 }),
+    }],
+    replicas: [{
+      region: 'us-east-1',
+    }],
+  });
+  ```
+
+* You can provide read capacity for GSIs where you want it to be different than the capacity specified for the replica.
+
+  ```typescript
+  new GlobalTable(tableStack, 'GlobalTable', {
+    tableName: 'FooTable',
+    partitionKey: {
+      name: 'FooHashKey',
+      type: AttributeType.STRING,
+    },
+    sortKey: {
+      name: 'FooRangeKey',
+      type: AttributeType.STRING,
+    },
+    billingMode: BillingMode.PROVISIONED,
+    write: Capacity.autoscaled({ max: 70 }),
+    globalSecondaryIndex: [{
+      indexName: 'UniqueGsiName',
+      partitionKey: {
+        name: 'FooRangeKey',
+        type: AttributeType.STRING,
+      },
+    }],
+    replicas: [{
+      region: 'us-east-1',
+      read: Capacity.autoscaled({ max: 70 }),
+      globalSecondaryIndexOptions: [
```

Comment:
> Was there a conscious decision between a list and a map?

Yes, I was just trying to keep it similar to how GSIs were defined. We can also have it like,

```
{
	'UniqueGsiName': {
		read: Capacity.fixed(55),
		contributorInsightsEnabled: false,
	}
}
```

Reply:
`read` and `contributorInsightsEnabled` are the only two props that can be configured for the GSI per replica. 

CFN resource: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-dynamodb-globaltable-replicaglobalsecondaryindexspecification.html

Reply:
```
{
	'UniqueGsiName': {
		read: Capacity.fixed(55),
		contributorInsightsEnabled: false,
	}
}
```

I kind of enjoy the above. Feels like less ceremony to me. No?

Reply:
Hey, I have just updated the `Replica GSI options` from list to map. I can see this can also be done for `GSI, Replica and LSI`. Let me know if that is what you would prefer. :) 

For me, it makes sense to have them as list since it conveys intent like, `list of global secondary indexes`, or `list of replicas`, but I am fine either way.